### PR TITLE
Issue 29/refactor gatsby starter default

### PR DESCRIPTION
**Pull request by saviomuc**
This PR fixes https://github.com/gatsbyjs/gatsby-starter-default/issues/29

It is quite big..
The goal of this PR is to get the template up to date to latest coding standards and clean up unused dependencies and files.

I tried not to be too strict (no enforced semicolons, basic eslint config, etc.)

### Config
- Remove gh-pages dependency
- Add prettier dependency and updated `format` script
- Add eslint dependency and `lint` script
- Add basic eslint config (Personally I'd prefer AirBnBs React default but they are super strict. Also on second thought it might make more sense to replicate the eslint config used in the main Gatbsy project, but then everybody has different opinions. Right now I don't have a clear solution.)
- Add .editorconfig which matches this (and Create React App) settings
- Add reference to official documentation in Readme

### Components
- Refactor components to be stateless components
- Remov unused import in components
- Unify export (export always in last line)
- Minor name changes to clarify purpose of each component

### Structure
- Move `css/typography` to `index/index.css` 

### What didn't work
Apparently this project needs a travis config for the CI to pass =/

### CODE REVIEW

**Comment by saviomuc** [comment]
@KyleAMathews PR is ready but fails for NODE_VERSION="0.12". Can you retrigger the build? Also is there a reason this specific version has to be tested? I feel like it would make more sense to test with Node > 8 as it's quite widespread and the most common stable.

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "rules": {
+    "linebreak-style": [
+      "error",
+      "unix"
+    ],
+    "quotes": [
+      "error",
+      "single"
+    ],
+    "semi": [
```
Comment (KyleAMathews) [inline .eslintrc.json]: I'd just remove all the stylistic suggestions. We don't care how people style their code.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+root = true
+
+[*]
+indent_style = space
```
Comment (KyleAMathews) [inline .editorconfig]: Just remove this whole file. We should keep to Gatsby's style while developing it but we shouldn't push that style on people's sites.

Comment (KyleAMathews) [inline .editorconfig]: I want to figure out how to move development of official starters to the Gatsby monorepo where we can handle enforcing coding style better.

File: src/html.js [hunk 3]
```diff
@@ -1,45 +1,48 @@
...
+  headComponents,
+  body,
+  postBodyComponents,
+}) => (
+  <html lang="en">
+    <head>
+      <meta charSet="utf-8" />
+      <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
+      <meta
+        name="viewport"
```
Comment (KyleAMathews) [inline src/html.js]: Could you just remove this file altogether? Gatsby has a default html.js file so generally people should just add things to their `<head>` with plugins and react-helmet (with gatsby-plugin-react-helmet) https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/cache-dir/default-html.js

**Comment by KyleAMathews** [comment]
You can just remove the travisci file as well actually. We don't really test anything and as I mentioned, I want to move building this starter to the main monorepo.

**Comment by KyleAMathews** [comment]
Also, could you just remove the eslintrc as well? I think a better solution is to copy CRA here and add the eslint-loader with their minimal setup. Want to create a PR to Gatsby itself doing that?

**Comment by saviomuc** [comment]
@KyleAMathews Thanks for the suggestions. I will update the PR as soon as possible and ping you. (Probably tomorrow)

Reagrding eslint: Can you create an issue in the Gatsby repository with your expectation and a "Need contribution" label or something? I am not too comfortable with CRA setup (need to look into it) but would be willing to take over as soon as I have time.

**Comment by KyleAMathews** [comment]
Created an issue here! https://github.com/gatsbyjs/gatsby/issues/1564

**Comment by saviomuc** [comment]
@KyleAMathews I updated the PR. Like it much better now. It's so clean =)

**Comment by saviomuc** [comment]
@KyleAMathews  Is there anything missin? Otherwise it would be great if you could merge the PR =)

**Comment by KyleAMathews** [comment]
Been vacationing! Will be back tomorrow though and merge then (can't merge on my phone due to Travis failure).

**Comment by KyleAMathews** [comment]
Thanks for the huge cleanup!!

**Pull request merged**